src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest.kt

```diff
@@ -0,0 +1,757 @@
+        object : ParameterizedTypeReference<String?>() {},
+        OFFENDER_NUMBER,
+      )
+      assertThatJsonFileAndStatus(responseV1, 200, "offender_detail_v1.1.json")
+      val httpEntityV1_1 = createHttpEntity(token, null, mapOf("version" to "1.1_beta"))
+      val responseV1_1 = testRestTemplate.exchange(
+        "/api/offenders/{nomsId}",
+        GET,
+        httpEntityV1_1,
+        object : ParameterizedTypeReference<String>() {},
+        OFFENDER_NUMBER,
+      )
+      assertThatJsonFileAndStatus(responseV1_1, 200, "offender_detail_v1.1.json")
+    }
+
+    @Test
+    fun testOffenderWithActiveRecallOffence() {
+      val token = authTokenHelper.getToken(VIEW_PRISONER_DATA)
+      val httpEntity = createHttpEntity(token, null)
+      val response = testRestTemplate.exchange(
+        "/api/offenders/{nomsId}",
+        GET,
+        httpEntity,
+        object : ParameterizedTypeReference<String?>() {},
+        "A1234AC",
+      )
+      assertThatJsonFileAndStatus(response, 200, "offender_detail_recall.json")
+    }
+
+    @Test
+    fun testOffenderWithInActiveRecallOffence() {
+      val token = authTokenHelper.getToken(VIEW_PRISONER_DATA)
+      val httpEntity = createHttpEntity(token, null)
+      val response = testRestTemplate.exchange(
+        "/api/offenders/{nomsId}",
+        GET,
+        httpEntity,
+        object : ParameterizedTypeReference<String?>() {},
+        "A1234AD",
+      )
+      assertThatJsonFileAndStatus(response, 200, "offender_detail_no_recall.json")
+    }
+
+    @Test
+    fun testOffenderInformationWithoutBooking() {
+      val token = authTokenHelper.getToken(VIEW_PRISONER_DATA)
+      val httpEntity = createHttpEntity(token, null)
+      val response = testRestTemplate.exchange(
+        "/api/offenders/{nomsId}",
+        GET,
+        httpEntity,
+        object : ParameterizedTypeReference<String?>() {},
+        "A1234DD",
+      )
+      assertThatJsonFileAndStatus(response, 200, "offender_detail_min.json")
+    }
+
+    @Test
+    fun testOffenderNotFound() {
+      val token = authTokenHelper.getToken(VIEW_PRISONER_DATA)
+      val httpEntity = createHttpEntity(token, null)
+      val response = testRestTemplate.exchange(
+        "/api/offenders/{nomsId}",
+        GET,
+        httpEntity,
+        object : ParameterizedTypeReference<String?>() {},
+        "B1234DD",
+      )
+      assertThatStatus(response, 404)
+    }
+
+    @Test
+    fun testFullOffenderInformation_WithAliases() {
+      val token = authTokenHelper.getToken(VIEW_PRISONER_DATA)
+      val httpEntity = createHttpEntity(token, null)
+      val response = testRestTemplate.exchange(
+        "/api/offenders/{nomsId}",
+        GET,
+        httpEntity,
+        object : ParameterizedTypeReference<String?>() {},
+        "A1234AI",
+      )
+      assertThatJsonFileAndStatus(response, 200, "offender_detail_aliases.json")
+    }
+  }
+
+  @Nested
+  @DisplayName("GET api/offenders/{offenderNo}/incidents")
```

Comment:
OffenderIncidents class tests access to the updated endpoint (all others are just pulled in from the java class)